### ISSUE TITLE
Some elements lack programmatic labels

### ISSUE THREAD

**Issue by SteelWagstaff** [pressbooks/pressbooks#2433]
## Export page
These should be added here: https://github.com/pressbooks/pressbooks/blob/dev/inc/modules/export/class-table.php
- [x] the checkboxes in the first column of the table on the Export page do not have programmatic labels. As a result these are sometimes communicated without a label at all, and sometimes they are labelled by the column header (Select All), neither of which communicate correct information to the user. 
![Screenshot from 2021-10-12 15-32-31](https://user-images.githubusercontent.com/13485451/137037835-214df624-5a9d-4292-9c10-4035d22a67a2.png)

- [x] checkboxes in the Pin column do not clearly communicate which document the user would be pinning by checking the box. Especially since the file name is not a header cell in this case, the label itself should provide this specificity (i.e. “Pin Accessibility-Audit-Demo 1629218879.pdf”)
![Screenshot from 2021-10-12 15-32-53](https://user-images.githubusercontent.com/13485451/137037833-98f40786-de66-4406-a74c-563a6fe08d5d.png)

## Book Info page
https://github.com/pressbooks/pressbooks/blob/dev/inc/admin/metaboxes/namespace.php
- [x] the Choose File buttons on the Contributors page and Edit Book Information page do not have a programmatic label. While the text that follows is grouped with the element and announced by assistive technologies, a programmatic label for the input is technically missing.
![Screenshot from 2021-10-12 15-36-09](https://user-images.githubusercontent.com/13485451/137038182-550eeedc-53b6-4406-9d28-2cd6f6d90636.png)
We appear to have buttons inside of buttons, and the container button lacks a label to help users of AT understand what the button does. WHY?
![Screenshot from 2021-10-19 09-58-18](https://user-images.githubusercontent.com/13485451/137957223-6bf6399c-6e41-48e1-91b4-c63b9bbb2f6a.png)
- [x] the Primary Subject input on the Edit Book Information page is not associated programmatically with its visual label. Instead of the label, the input value is communicated to assistive technologies. 
![Screenshot from 2021-10-12 15-36-43](https://user-images.githubusercontent.com/13485451/137038220-5bbf3bd1-ec27-4150-ac3e-4a5ad9e565b9.png)
- [x] The purpose of text editors is not clear to assistive technologies. These essentially function as large text boxes and so should have a programmatic label. Because they are embedded within an iframe, this can be accomplished through the iframe title. The boxes do have titles but the titles don’t include the label. For example the About section on the Cover Generator page is labelled “Rich Text Area. Press Control-Option-H for help”. This title should also include the purpose of the editor (About).
![Screenshot from 2021-10-12 15-37-24](https://user-images.githubusercontent.com/13485451/137038275-3f0501a9-28c4-46d3-ac6a-43a666ee5208.png)

**Comment by SteelWagstaff** [comment]
Lisa from the IDRC recommends two articles which she thinks can help resolve the issue with inputs such as the Author input on the Book Edit page: 
https://www.24a11y.com/2019/select-your-poison/
https://www.24a11y.com/2019/select-your-poison-part-2/
Part 2 proposes a pattern that is very similar to what you currently have (the buttons above combobox input in the Multiselect section). Based on the user research they did it performed really well with users so I think it will make for a very accessible solution.

**Comment by SteelWagstaff** [comment]
Can potentially replace this issue with findings from latest audit. FYI @MichelleWeremczuk

**Comment by greatislander** [comment]
Some of this is now resolved but would be valuable to split out the remaining issues into their own tickets, @SteelWagstaff @cmurtagh.

**Comment by greatislander** [comment]
The issue with TinyMCE areas is captured in pressbooks/private#1706. The rest of these issues are resolved as far as I can tell.
